exo-go/src/app_runner/index.go

```diff
@@ -90,26 +91,29 @@ func (a *AppRunner) Start() error {
 	if err != nil {
 		return err
 	}
+	var wg sync.WaitGroup
+	var onlineTextRegex *regexp.Regexp
 	for role, onlineText := range onlineTexts {
-		if err = a.waitForOnlineText(process, role, onlineText); err != nil {
+		wg.Add(1)
+		onlineTextRegex, err = regexp.Compile(fmt.Sprintf("%s.*%s", role, onlineText))
```

Comment:
Moved this regular expression compilation up here so we can deal with the error if there is one and now the goroutine doesn't need to somehow pass an error